app/src/org/commcare/logic/AndroidFormController.java

```diff
@@ -96,4 +99,17 @@ public String getSerializedFormIndex() {
             return null;
         }
     }
+
+    private FormIndex deserializeFormIndex(String serializedFormIndex) {
```

Comment:
can we perform the deserialization at the same time when we retrive the form index in `FormEntryActivity`. Today the logic to retrive the saved value of the index seems to be split into multiple place which is not ideal.

Reply:
Very true, I wanted to keep the `serialization` and `deserialization` logic together, but this is method doesn't rely on the state of the class